lib/src/layer/polyline_layer.dart

```diff
@@ -39,66 +79,292 @@ class Polyline {
-      borderStrokeWidth,
-      borderColor,
-      gradientColors,
-      colorsStop,
-      isDotted,
-      strokeCap,
-      strokeJoin,
-      useStrokeWidthInMeter);
+        strokeWidth,
+        color,
+        borderStrokeWidth,
+        borderColor,
+        gradientColors,
+        colorsStop,
+        isDotted,
+        strokeCap,
+        strokeJoin,
+        useStrokeWidthInMeter,
+      );
 }
 
 @immutable
-class PolylineLayer extends StatelessWidget {
-  final List<Polyline> polylines;
-  final bool polylineCulling;
+@optionalTypeArgs
+class PolylineLayer<TapKeyType extends Object> extends StatelessWidget {
+  final List<Polyline<TapKeyType>> polylines;
+
+  /// Called when a tap is detected on any [Polyline] with a defined
+  /// [Polyline.tapKey]
+  ///
+  /// Individual [Polyline]s may have their own [Polyline.onTap] callback
+  /// defined, regardless of whether this is defined.
+  ///
+  /// See [nonTappablesObscure] and [tappablesObscure] to set behaviour when a
+  /// tap is over multiple overlapping [Polyline]s.
+  final PolylineLayerOnTap<TapKeyType>? onTap;
+
+  /// Called when a long press is detected on any [Polyline] with a defined
+  /// [Polyline.tapKey]
+  ///
+  /// Individual [Polyline]s may have their own [Polyline.onLongPress] callback
+  /// defined, regardless of whether this is defined.
+  ///
+  /// See [nonTappablesObscure] and [tappablesObscure] to set behaviour when a
+  /// tap is over multiple overlapping [Polyline]s.
+  final PolylineLayerOnTap<TapKeyType>? onLongPress;
+
+  /// Called when a secondary tap/alternative click is detected on any [Polyline]
+  /// with a defined [Polyline.tapKey]
+  ///
+  /// Individual [Polyline]s may have their own [Polyline.onSecondaryTap]
+  /// callback defined, regardless of whether this is defined.
+  ///
+  /// See [nonTappablesObscure] and [tappablesObscure] to set behaviour when a
+  /// tap is over multiple overlapping [Polyline]s.
+  final PolylineLayerOnTap<TapKeyType>? onSecondaryTap;
+
+  /// The number of pixels away from the visual line (including any width and
+  /// outline) in which a tap should still register as a tap on the line
+  ///
+  /// Applies to both [onTap] and every [Polyline.onTap].
+  ///
+  /// Defaults to 3.
+  final double onTapTolerance;
+
+  /// Whether a non-tappable [Polyline] should prevent taps from being handled
+  /// on all [Polyline]s beneath it, at overlaps
+  ///
+  /// "Tappable" means that either or both of [Polyline.onTap] and
+  /// [Polyline.tapKey] has been defined.
+  ///
+  /// Applies to both [onTap] and every [Polyline.onTap].
+  ///
+  /// Defaults to `true`.
+  final bool nonTappablesObscure;
+
+  /// Whether a tappable [Polyline] should prevent taps from being handled
+  /// on all [Polyline]s beneath it, at overlaps
+  ///
+  /// "Tappable" means that either or both of [Polyline.onTap] and
+  /// [Polyline.tapKey] has been defined.
+  ///
+  /// If `false`, then [onTap] becomes redundant to [Polyline.onTap].
```

Comment:
Whoops, that should probably be `true` not `false`.

```suggestion
  /// If `true`, then [onTap] becomes redundant to [Polyline.onTap].
```